### PR TITLE
Update core and blocks to SvelteKit v2

### PR DESCRIPTION
**Pull request by mcous**
## Overview

Excluding lockfile:

```
28 files changed, 163 insertions(+), 162 deletions(-)
```

I was working on updating SvelteKit in the downstream app, and immediately hit a couple problems from Prime. I decided to do the update here first to shake everything out.

SvelteKit v2 updates to Vite v5, which is much more strict about CJS vs ESM. We had two imports in Prime that pretended CJS deps were ESM, which Vite v5 refused to bundle:

- PrismJS
- Lodash

We can no longer implicitly destructure an object exported as a CJS default export:

```ts
import { highlightElement, plugins } from 'prismjs';
```

Instead, we must be explicit:

```ts
import Prism from 'prismjs'
const { highlightElement, plugins } = Prism
```

This is good, because it stops hiding imports where we thought we were getting tree-shaking but weren't. For Prism, I did the change above. For lodash, I swapped it out for lodash-es, which was on my todo list anyway.

## Change log

- Update all dependencies, excect:
  - Didn't touch legacy at all
  - Blocks
      - mapbox
  - Storybook
      - sveltekit (SB hasn't caught up to Vite v5 yet)
- Swap lodash for lodash-es
- Make minimal fixes to get builds / checks / dev environments working

## Review requests

Before merging, I'll be `pnpm link`'ing these to app to make sure things keep functioning.

- Try out the dev sites if you can
- Review source code changes with paranoia

### CODE REVIEW

**Review by DTCurrie — APPROVED** [review]
I was real skeptical about that PrismJs import.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,6 +55,9 @@ jobs:
       - name: Install 🔧
         uses: viamrobotics/js-config/.github/actions/setup@0006127efb3f62669f662ca31c555a49a034935e
 
+      - name: Build 🏗️
+        run: pnpm run build
+
```
Comment (ethanlook) [inline .github/workflows/ci.yml:60]: Should we just combine this with the above?

**Review by ethanlook — APPROVED** [review]
Why did tailwind reorder?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mcous) [reply .github/workflows/ci.yml:60]: I'm here for it

**Comment by mcous** [comment]
> Why did tailwind reorder?

The new order is definitely correct, so I'm not sure why it was improperly ordered in the first place. Since the end result is correct I didn't really look that deeply into it

**Comment by mcous** [comment]
Synced with main, updated stragglers, and ran a test with `pnpm link` for my local copy of the app. Did not experience any issues in either Vue or Svelte, so going to merge this as soon as CI is green

**Pull request merged**